### PR TITLE
system: Introduce ipset

### PR DESCRIPTION
**Pull request by ssahani**
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
This work enables to configure ipset . ipset is a companion application for the iptables Linux firewall.
It allows you to setup rules to quickly and easily block a set of IP addresses among other things.

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
system
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
"ansible 2.8.0.dev0"
```

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

       - ipset:                                                                                                                                                                                
             ipset_name: test3                                                                                                                                                                 
             hash_type:  hash:ip                                                                                                                                                               
             keys: "10.65.208.22 10.65.208.26 10.65.208.28"                                                                                                                                    
             state: create                                                                                                                                                                     
       - ipset:                                                                                                                                                                                
             ipset_name: test3                                                                                                                                                                 
             keys: "10.65.208.22 10.65.208.26 10.65.208.28"                                                                                                                                    
             state: add                                                                                                                                                                        
       - ipset:                                                                                                                                                                                
             ipset_name: test3                                                                                                                                                                 
             keys: "10.65.208.22"                                                                                                                                                              
             state: delete                                                                                                                                                                     
       - ipset:                                                                                                                                                                                
             ipset_name: test3                                                                                                                                                                 
             state: destroy                                                                                                                                                                    
                                  
```
Signed-off-by: Susant Sahani <susant@redhat.com>

### CODE REVIEW

File: lib/ansible/modules/system/ipset.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['stableinterface'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: ipset
+short_description: Automateds ipset
```
Comment (jbenden) [inline lib/ansible/modules/system/ipset.py]: Typo. Should be "Automates ipset".

File: lib/ansible/modules/system/ipset.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+description:
+    - Allows you to setup rules to quickly and easily block of setup ip addresses.
```
Comment (jbenden) [inline lib/ansible/modules/system/ipset.py]: Typo. "Allows you to setup rules to quickly and easily block sets of ip addresses."

File: lib/ansible/modules/system/ipset.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        description:
+            - creating a new "set" of network addresses.
+    hash_type:
+       description: What kind of set
+       choices: [ "hash:net", "hash:ip", "hash:mac", "hash:ip,mac", "hash:net,net", "hash:ip,port",
+                  "hash:net,port", "hash:ip,port,ip", "hash:ip,port,net", "hash:ip,mark", "hash:net,port,net",
+                  "hash:net,iface" ]
+    keys:
+        description:
+            - Space separated list of kesy according to the hash type.
```
Comment (jbenden) [inline lib/ansible/modules/system/ipset.py]: Typo. "Space separated list of keys according to the hash type."

Comment (ssahani) [reply lib/ansible/modules/system/ipset.py]: Thanks for the review @jbenden . Updated.

File: lib/ansible/modules/system/ipset.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        argument_spec=dict(
+            ipset_name=dict(required=True, type='str'),
+            hash_type=dict(choices=['hash:net', 'hash:ip', 'hash:mac', 'hash:ip,mac', 'hash:net,net',
+                                    'hash:ip,port', 'hash:net,port', 'hash:ip,port,ip', 'hash:ip,port,net',
+                                    'hash:ip,mark', 'hash:net,port,net', 'hash:net,iface'],
+                           required=False, default=None, type='str'),
+            keys=dict(required=False, default=None, type='str'),
+            state=dict(choices=['add', 'delete', 'destroy', 'create'], required=True),
+        ),
+        supports_check_mode=True
```
Comment (maxamillion) [inline lib/ansible/modules/system/ipset.py]: This claims support of `check_mode` but then never checks for it in order to call `module.exit_json` without actually making changes.

File: lib/ansible/modules/system/ipset.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        module.fail_json(msg='hash_type cannot be None when state is create')
+
+    if keys is None:
+        if state == 'add' or state == 'delete':
+            module.fail_json(msg='Keys cannot None when state is add or delete')
+
+    ipset = IpsetModule(module)
+    result = ipset.configure_ipset()
+
+    module.exit_json(changed=result)
```
Comment (maxamillion) [inline lib/ansible/modules/system/ipset.py:176]: `result` appears to be just a return code, is there any desire to have a more verbose message to provide context to the user about what took place?

File: lib/ansible/modules/system/ipset.py (hunk 4 above)
Comment (ssahani) [reply lib/ansible/modules/system/ipset.py]: ya right should be false . thanks

File: lib/ansible/modules/system/ipset.py (hunk 5 above)
Comment (ssahani) [reply lib/ansible/modules/system/ipset.py:176]: May be for improvement PR.

**Pull request closed without merging**